android/src/main/java/com/reactlibrary/QonversionModule.java

```diff
@@ -65,16 +67,31 @@ public String getName() {
     }
 
     @ReactMethod
-    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion) {
-        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(getCurrentActivity().getApplication()).edit();
+    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion,final Promise promise) {
+        Activity currentActivity = getCurrentActivity();
+        if(currentActivity == null){
+            QonversionError qonversionError = new QonversionError(QonversionErrorCode.UnknownError, "Android current activity is null, cannot perform the process.");
```

Comment:
I think will be better to move this error generation code to a separate function and reuse it.

Reply:
Done